_content/tour/por/generics-type-constraints.article

```diff
@@ -1,68 +1,66 @@
+exemplos nesta seção do Tour.
 
 .html generics-video.html
 
-** Code Review
+** Revisão de Código
 
-- *Example* *1*: Type based constraint
-- *Example* *2*: Predeclared type constraint "comparable"
-- *Example* *3*: Mix type and behavior constraints
+- *Exemplo* *1*: Restrição baseada em tipo
+- *Exemplo* *2*: Restrição de tipo predefinido "comparable" (comparável)
+- *Exemplo* *3*: Mistura de restrições de tipo e comportamento
 
 .play generics/type-constraints/example1.go
 .play generics/type-constraints/example2.go
 .play generics/type-constraints/example3.go
 
-** Explained
+** Explicativo
 
-Generic functions create a new type of constraint that can’t be resolved by declaring
-a method set of behavior.
+Funções genéricas criam um novo tipo de restrição que não pode ser resolvida declarando
+um conjunto de métodos de comportamento.
 
     func Add[T ???](v1 T, v2 T) T {
         return v1 + v2
     }
 
-Here is a generic function that wants to accept two values of some type T, add them
-together, and then return the sum back to the caller. This is an interesting problem
-because the compiler needs to constrain the call to the function for only values that
-can be used in an add operation. Currently there is no mechanic for declaring this
-kind of constraint.
+Aqui está uma função genérica que deseja aceitar dois valores de algum tipo T, somá-los
+e então retornar a soma para quem chamou. Este é um problema interessante porque o compilador 
+precisa restringir a chamada à função apenas para valores que podem ser usados em uma operação de adição. 
+Atualmente não há um mecanismo para declarar este tipo de restrição.
 
-The decision was to continue to use the interface to declare the constraint and add
-something new.
+A decisão foi continuar a usar a interface para declarar a restrição e adicionar algo novo.
 
     type addOnly interface {
         string | int | int8 | int16 | int32 | int64 | float64
     }
 
-You can declare an interface that defines a set of types that form the constraint. Then
-apply this interface to the generic function.
+Você pode declarar uma interface que define um conjunto de tipos que formam a restrição. Para então
+aplicar esta interface à função genérica.
 
     func Add[T addOnly](v1 T, v2 T) T {
         return v1 + v2
     }
 
-Now the compiler can validate that the set of types is compliant with the operations
-the function needs to perform against values of those types. When the interface is
-using the built-in types, the interfaces are reusable across packages. When the list
-of types represent user-defined types from the package, You must remember these
-generic functions are bound to the packages types and nothing more.
+Agora o compilador pode validar que o conjunto de tipos é compatível com as operações
+que a função precisa realizar com valores daqueles tipos. Quando a interface é
+usando os tipos embutidos, as interfaces são reutilizáveis entre pacotes. Quando a lista
```

Comment:
I see "embutido" as a better translation of embedded than built-in. When translating some articles I had difficulty to translate built-in too, I thought in "primitivo ou básico", but none of them fit every situation so I didn't translate.
```suggestion
que a função precisa realizar com valores daqueles tipos. Quando a interface está
usando os tipos built-in, as interfaces são reutilizáveis entre pacotes. Quando a lista
```